tests/test_gmsh.py

```diff
@@ -1,35 +1,99 @@
+# POSSIBILITY OF SUCH DAMAGE.
+
 import os
 
 import unittest
 
 import numpy as np
 
-import modmesh
+import modmesh as mm
+
 
+class GmshTB(unittest.TestCase):
+    TESTDIR = os.path.abspath(os.path.dirname(__file__))
+    DATADIR = os.path.join(TESTDIR, "data")
 
-class GmshTC(unittest.TestCase):
 
+class GmshTriangleTC(GmshTB):
     def test_gmsh_parsing(self):
-        path = os.path.join(os.path.abspath(os.path.dirname(__file__)),
-                            "data", "gmsh_triangle.msh")
+        path = os.path.join(self.DATADIR, "gmsh_triangle.msh")
 
         data = open(path, 'rb').read()
-        gmsh_instance = modmesh.core.Gmsh(data)
-        blk = gmsh_instance.to_block()
+        gmsh = mm.Gmsh(data)
+        blk = gmsh.to_block()
 
         # Check nodes information
         self.assertEqual(blk.nnode, 4)
 
         # Due to ghost cell and ghost node had been created, the real body
-        # had been shifted and start with index 3
-        np.testing.assert_almost_equal(blk.ndcrd.ndarray[3:, :].tolist(),
+        # had been shifted and start with index 3 (number of ghost)
+        ngst = blk.ngstcell
+        self.assertEqual(ngst, 3)
+        np.testing.assert_almost_equal(blk.ndcrd.ndarray[ngst:, :].tolist(),
                                        [[0.0, 0.0],
                                         [-1.0, -1.0],
                                         [1.0, -1.0],
                                         [0.0, 1.0]])
         # Check cells information
         self.assertEqual(blk.ncell, 3)
-        self.assertEqual(blk.cltpn.ndarray[3:].tolist(), [4, 4, 4])
-        self.assertEqual(blk.clnds.ndarray[3:, :4].tolist(), [[3, 0, 1, 2],
-                                                              [3, 0, 2, 3],
-                                                              [3, 0, 3, 1]])
+        self.assertEqual(blk.cltpn.ndarray[ngst:].tolist(), [4, 4, 4])
+        self.assertEqual(blk.clnds.ndarray[ngst:, :4].tolist(),
+                         [[3, 0, 1, 2],
+                          [3, 0, 2, 3],
+                          [3, 0, 3, 1]])
+
+
+class GmshRectangularTC(GmshTB):
+    def setUp(self):
+        # Read the Gmsh mesh file.
+        path = os.path.join(self.DATADIR, "rectangle.msh")
+        with open(path, 'rb') as fobj:
+            data = fobj.read()
+        # Create the Gmsh object.
+        gmsh = mm.Gmsh(data)
+        # Convert the Gmsh object to StaticMesh object.
+        self.blk = gmsh.to_block()
+
+    def test_shape(self):
+        blk = self.blk
+        # Test for shape data.
+        self.assertEqual(blk.ndim, 2)
+        self.assertEqual(blk.nbound, 40)
+        self.assertEqual(blk.nnode, 104)
+        self.assertEqual(blk.nface, 309)
+        self.assertEqual(blk.ncell, 206)
+        self.assertEqual(blk.ngstnode, 40)
+        self.assertEqual(blk.ngstface, 80)
+        self.assertEqual(blk.ngstcell, 40)
+
+    @unittest.expectedFailure
```

Comment:
@j8xixo12 , it seems that we have a bug.  I will create the issue later.

The mesh is all triangles.  All cell types should be 4, but it fails.  Some becomes 2 (line/edge).